tools/app/app/lib/hooks/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { usePathname } from 'next/navigation'
+import { useCallback, useEffect, useMemo, useRef } from 'react'
+import { findParentCard } from '../utils'
+import { useProject } from '../api'
+
+export function useCardKey() {
```

Comment:
In my opinion, if we are using the app router correctly, the page or layout should retrieve the card key from the url parameters and provide it to its child components. Otherwise there might be something wrong with the logical structure of the app. Also, we happen to change the routing, this hook will break.